types/string-algorithms/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for string-algorithms 1.0
+// Project: https://github.com/burgaard/string-algorithms
+// Definitions by: lionel-rowe <https://github.com/DefinitelyTyped>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.0
+
+export function search(text: string, term: string): number[];
+
+export function radixSort<T>(
+    ...args: T extends number[]
+        ? [entries: T[], getEntry?: (el: T) => number[]]
+        : [entries: T[], getEntry: (el: T) => number[]]
+): number[][];
+
+export function suffixArray(s: number[] | string, terminator?: number): number[];
+
+export function longestCommonPrefix(sequence: string[], suffixArray: number[]): number[];
+
+export function longestCommonSubstring(strings: string[], indexMap?: 'log' | 'linear' | StringIndexMap): string[];
+
+export class StringIndexMap {
```

Comment:
in terms of TS, isn't that class supposed to be `abstract`?

> Concrete implementations provide different compromises between O(1) and O(log(k)) lookup times versus O(n) and O(k) space requirements. Extend this class to implement custom mappings from string positions to substring indices with different runtime/space tradeoffs than the two pre-defined implementations.

https://github.com/burgaard/string-algorithms#class-stringindexmap

Reply:
Yes, thanks for spotting! The class can be instantiated, but calling any of its methods just throws errors, so practically speaking it's abstract. Fixed now.